### DISCUSSION TITLE
How are the icons for GPX-Tracks determined?

### DISCUSSION THREAD

**Discussion by hajo62** [bpatrik/pigallery2#667] [Q&A] [answered]
I have a folder with two tracks enclosed. When I look at the map, one is marked as walking, the second with a gray dot.
How is the icon determined? 
I haven't found a category in the track files...
<img width="418" alt="image" src="https://github.com/bpatrik/pigallery2/assets/15800980/68bf9eef-96ff-4c57-b3a2-f0b8ab14aeaa">

**Comment by bpatrik** [comment]
Sofar the rules are hard coded here:


https://github.com/bpatrik/pigallery2/blob/master/src/frontend/app/ui/gallery/map/lightbox/lightbox.map.gallery.component.ts#L116-L137

-- Sorry for being brief, sent from my phone.

**Reply by hajo62** [reply]
The info is grabbed from the filename or from some metadata?

**Comment by bpatrik** [comment, marked as answer]
Yes. If I recall correctly. It's looking for a Name tag.

Garmin and runkeeper sets those correctly

-- Sorry for being brief, sent from my phone.

**Comment by hajo62** [comment]
Yes, look's like the name tag is used. 
  THX

**Comment by bpatrik** [comment]
I made a change to make them configurable. Its not yet on the UI, but you can find them in the config.json Not super straight forward yet, but at least there is a way to change it.

**Comment by knalltuete86** [comment]
I like to use this discussion to add a related question:

I do have GPX Files exported from my Polar Watch. Example:

`<?xml version="1.0" encoding="UTF-8"?><gpx xmlns="http://www.topografix.com/GPX/1/1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.1" creator="Polar Vantage M" xsi:schemaLocation="http://www.topografix.com/GPX/1/1 http://www.topografix.com/GPX/1/1/gpx.xsd">
<metadata>
<name>MTB-Tour zur Quelle der Mosel</name>
<author><name>Polar</name></author>
<time>2023-09-01T18:26:53.061Z</time>
<bounds minlat="47.85348667" minlon="6.70669333" maxlat="47.89682" maxlon="6.89257"></bounds>
</metadata>
<trk><name>MTB-Tour zur Quelle der Mosel</name>
<trkseg>
<trkpt lat="47.89141833" lon="6.70678667"><ele>560.0</ele><time>2022-09-12T10:20:21.423Z</time></trkpt><trkpt ...`

The `name`-Tag within the `author`-Tag is used for the Popup on the Map when I do click on the marker.

I added another `name`-Tag including a Description within the `metadata`- and `trk`-Tag and added the corresponding Regex to the Configuration (e.g. `MTB`). But it does not change the Marker at all.

I also tried to remove the `name`-Tag from the `author` - but it doesn't work.

Does anyone have a sample file which do work (showing the correct icons based on the marker configuration)?

Regards

**Reply by knalltuete86** [reply]
Based on the sample GPX-files in the `demo` folder I can answer myself: The `name`-Tag of the single `trk` is used for determination of the marker. Example:

`<trk>
<name>Hiking</name>
<trkpt>...`

should work after new indexing.

Regards

**Reply by bpatrik** [reply]
Never tested in all imputs, but the code just looks for any `<name>Hiking</name>` tags. It uses the first that it finds. Then here is how they gets categorized:
![kép](https://github.com/bpatrik/pigallery2/assets/7512182/584daa37-91be-4fe9-8639-6eec558097f5)
